pkg/apis/core/validation/shoot.go

```diff
@@ -1985,6 +1985,20 @@ func ValidateKubeletConfig(kubeletConfig core.KubeletConfig, version string, fld
 		}
 	}
 
+	if v := kubeletConfig.MaxParallelImagePulls; v != nil {
+		path := fldPath.Child("maxParallelImagePulls")
+
+		if *v < 1 {
+			allErrs = append(allErrs, field.Invalid(path, *v, "value must be >= 1"))
+		}
+
+		if serializeImagePull := kubeletConfig.SerializeImagePulls; serializeImagePull != nil {
+			if *serializeImagePull && *v > 1 {
+				allErrs = append(allErrs, field.Forbidden(path, "configuring limit of image pulls is not available when kubelet's 'SerializeImagePulls' is set"))
```

Comment:
Nit: I would vote in error message to use the field names used in serialization instead of the field names in the go struct. 

```suggestion
				allErrs = append(allErrs, field.Forbidden(path, "maxParallelImagePulls cannot be larger than 1 when serializeImagePulls is set to true"))
```